### PR TITLE
SPR-50 Individual request page

### PR DESCRIPTION
**Pull request by dbarkowsky**
# MERGE AFTER SPR-49

# Description

This PR includes the following proposed change(s):

- Added more components that match style for BCGov
- Created a page for individual requests -> form with locked, prepopulated values where status can be updated
- Styling changes for responsiveness

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change
- [x] Dependencies added/removed

## Aspect(s) of Project Affected
- [x] Frontend
- [ ] API
- [ ] Database
- [ ] Workflows
- [ ] Documentation

## Best Practices Checklist
- [x] I have performed a self-review of my own code
- [x] My code follows the Airbnb React Style Guidelines
- [ ] Documentation has been updated to reflect my changes
- [x] New and existing tests pass locally with my changes

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.
Tested locally through use.

### CODE REVIEW

File: app/src/components/bcgov/ButtonStyles.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    borderRadius: '4px',
+    padding: '12px 24px',
+    textAlign: 'center',
+    textDecoration: 'none',
+    display: 'inline-block',
+    fontSize: '14px',
+    fontWeight: '700',
+    letterSpacing: '1px',
+    cursor: 'pointer',
+  '&:hover': {
```
Comment (akroon3r) [inline app/src/components/bcgov/ButtonStyles.ts:15]: same feedback as the other PR

File: app/src/components/bcgov/ButtonStyles.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    color: bcgov.white
+  }
+}
+
+export const buttonStyles = {
+  primary: {
+      ...commonButtonStyle,      
+      backgroundColor: bcgov.primary,
+      color: bcgov.white,
+    disabled: {
```
Comment (akroon3r) [inline app/src/components/bcgov/ButtonStyles.ts:26]: same as above

File: app/src/utils/convertState.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
+export enum RequestStates {
+  DELETED,
```
Comment (akroon3r) [inline app/src/utils/convertState.ts:2]: Lets follow up with Genita tomorrow to double check that this list of states is correct.

File: app/package.json [hunk 4]
```diff
@@ -11,8 +11,16 @@
     "cypress:e2e": "cypress run --e2e"
   },
   "dependencies": {
+    "@bcgov/bc-sans": "^1.0.1",
+    "@emotion/react": "^11.11.0",
```
Comment (BradyMitch) [inline app/package.json:15]: You should remove the caret symobols `^` from your package.json dependencies and run the command `npm config set save-exact true` so this doesnt happen again.

File: app/src/components/custom/tables/RequestsTable.tsx [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+          ? <TableRow><CustomTableCell>No requests available.</CustomTableCell></TableRow>
+          : data.map((row, index) => (
+            <TableRow
+              key={row._id}
+              sx={{ '&:last-child td, &:last-child th': { border: 0 }, backgroundColor: index % 2 === 0 ? bcgov.white : bcgov.backgroundSecondary }}
+            >
+              <CustomTableCell>{`${row.firstName} ${row.lastName}`}</CustomTableCell>
+              <CustomTableCell>{row.itemsPurchased.join('; ')}</CustomTableCell>
+              <CustomTableCell>{`$${row.totalCost}`}</CustomTableCell>
+              <CustomTableCell>{new Date(row.purchaseDate).toLocaleDateString()}</CustomTableCell>
```
Comment (BradyMitch) [inline app/src/components/custom/tables/RequestsTable.tsx:49]: Not sure if you want this but something to look into for date formatting using functionality built into JavaScript: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat

File: app/src/main.tsx [hunk 6]
```diff
@@ -1,7 +1,6 @@
 import React from 'react'
 import ReactDOM from 'react-dom/client'
 import App from './App'
-import './index.css'
 
 ReactDOM.createRoot(document.getElementById('root') as HTMLElement).render(
```
Comment (BradyMitch) [inline app/src/main.tsx:5]: Not sure if eslint is giving you any trouble with this but I believe this is the way React wants you to createRoot in version 18: https://github.com/BradyMitch/LaunchPad/blob/main/react-express-postgres/frontend/src/main.tsx

File: app/src/components/custom/tables/RequestsTable.tsx (hunk 5 above)
Comment (dbarkowsky) [reply app/src/components/custom/tables/RequestsTable.tsx:49]: Interesting option I didn't know was out there.

File: app/package.json (hunk 4 above)
Comment (dbarkowsky) [reply app/package.json:15]: Will do.

File: app/src/components/bcgov/ButtonStyles.ts (hunk 1 above)
Comment (dbarkowsky) [reply app/src/components/bcgov/ButtonStyles.ts:15]: Should be covered from the other PR. Will merge to update this.

File: app/src/components/bcgov/ButtonStyles.ts (hunk 2 above)
Comment (dbarkowsky) [reply app/src/components/bcgov/ButtonStyles.ts:26]: Same update as above.

File: app/src/main.tsx (hunk 6 above)
Comment (dbarkowsky) [reply app/src/main.tsx:5]: I believe this was what Vite spawned on its own. I think I'll leave it as is for now, but if I end up needing to store the element, I'll convert it to that.

File: .gitignore [hunk 7]
```diff
@@ -13,6 +13,7 @@ dist
 dist-ssr
 *.local
 .env
+package-lock.json
```
Comment (BradyMitch) [inline .gitignore:16]:
```suggestion
**/package-lock.json
```

Comment (BradyMitch) [reply .gitignore:16]: You also need to `git rm` the package-lock files since they have already been added to the repo.

Comment (dbarkowsky) [reply .gitignore:16]: Seems to have removed them according to the diff. 
Is there a benefit to using the ** if the gitignore picks up those files without it?

Comment (BradyMitch) [reply .gitignore:16]: If the ignore file picks them up then ignore me.

Comment (BradyMitch) [reply .gitignore:16]: If the files are still in the repo after the merge you'll have to use the git rm command

**Review by BradyMitch — APPROVED** [review]

**Pull request merged**